dask_kubernetes/tests/test_core.py

```diff
@@ -29,59 +29,61 @@
 
 
 @pytest.fixture
-def api():
-    ClusterAuth.load_first()
-    return kubernetes.client.CoreV1Api()
+async def api():
+    await ClusterAuth.load_first()
+    yield kubernetes.client.CoreV1Api()
 
 
 @pytest.fixture
-def ns(api):
-    name = "test-dask-kubernetes" + str(uuid.uuid4())[:10]
+async def ns(api):
+    name = "test-dask-kubernetes-" + str(uuid.uuid4())[:4]
     ns = kubernetes.client.V1Namespace(
         metadata=kubernetes.client.V1ObjectMeta(name=name)
     )
-    api.create_namespace(ns)
+    await api.create_namespace(ns)
     try:
         yield name
     finally:
-        api.delete_namespace(name)
+        await api.delete_namespace(name)
 
 
 @pytest.fixture
-def pod_spec(image_name):
+async def pod_spec(image_name):
     yield make_pod_spec(
         image=image_name, extra_container_config={"imagePullPolicy": "IfNotPresent"}
     )
 
 
 @pytest.fixture
-def clean_pod_spec(pod_spec):
+async def clean_pod_spec(pod_spec):
     yield clean_pod_template(pod_spec)
 
 
 @pytest.fixture
-def cluster(pod_spec, ns, loop):
-    with KubeCluster(pod_spec, loop=loop, namespace=ns) as cluster:
+async def cluster(pod_spec, ns):
+    async with KubeCluster(pod_spec, namespace=ns, asynchronous=True) as cluster:
         yield cluster
 
 
 @pytest.fixture
-def client(cluster):
-    with Client(cluster) as client:
+async def client(cluster):
+    async with Client(cluster) as client:
         yield client
 
 
-def test_versions(client):
+@pytest.mark.asyncio
+async def test_versions(client):
     client.get_versions(check=True)
 
 
-def test_basic(cluster, client):
+@pytest.mark.asyncio
+async def test_basic(cluster, client):
     cluster.scale(2)
     future = client.submit(lambda x: x + 1, 10)
     result = future.result()
     assert result == 11
 
-    while len(cluster.scheduler.workers) < 2:
+    while len(cluster.workers) < 2:
         sleep(0.1)
```

Comment:
These two checks are actually different.

If you want to wait for two workers on the cluster, then you can just do `await cluster`

However if you want to wait until the scheduler has seen two workers, then you might do `cluster.scheduler_info["workers"]`.  This value is updated by a communication channel regularly.